File: articles/inorganic-chemistry-diversity-non-carbon-systems/julia/inorganic_kernel.jl
# Inorganic chemistry numerical kernel in Julia.

function oxidation_state(total_charge, known_contribution, unknown_atom_count)
    return (total_charge - known_contribution) / unknown_atom_count
end

function cfse(t2g_electrons, eg_electrons, delta_o)
    return t2g_electrons * (-0.4 * delta_o) + eg_electrons * (0.6 * delta_o)
end

function spin_only_moment(unpaired_electrons)
    return sqrt(unpaired_electrons * (unpaired_electrons + 2.0))
end

function tolerance_factor(rA, rB, rX)
    return (rA + rX) / (sqrt(2.0) * (rB + rX))
end

println("Mn_in_KMnO4_OS=", round(oxidation_state(0.0, -7.0, 1.0), digits=6))
println("octahedral_d3_CFSE=", round(cfse(3.0, 0.0, 1.0), digits=6))
println("spin_only_d3=", round(spin_only_moment(3.0), digits=6))
println("tolerance_factor=", round(tolerance_factor(1.60, 0.60, 1.40), digits=6))

File: articles/inorganic-chemistry-diversity-non-carbon-systems/fortran/inorganic_kernel.f90
program inorganic_kernel
  implicit none

  real(8) :: mn_os, d3_cfse, spin_moment, tolerance

  mn_os = (0.0d0 - (-7.0d0)) / 1.0d0
  d3_cfse = 3.0d0 * (-0.4d0 * 1.0d0) + 0.0d0 * (0.6d0 * 1.0d0)
  spin_moment = sqrt(3.0d0 * (3.0d0 + 2.0d0))
  tolerance = (1.60d0 + 1.40d0) / (sqrt(2.0d0) * (0.60d0 + 1.40d0))

  print *, "Mn_in_KMnO4_OS=", mn_os
  print *, "octahedral_d3_CFSE=", d3_cfse
  print *, "spin_only_d3=", spin_moment
  print *, "tolerance_factor=", tolerance

end program inorganic_kernel
